[MOB-1753] StatusBadge, ChatBadge 구현

bonoogi opened the pull request:
## 어떤 PR인가요?

베지어 컴포넌트중 StatusBadge, ChatBadge를 구현하는 PR입니다.

## 작업 내용

- StatusBadge와 그에 대응하는 StatusBadgeConfiguration을 구현합니다.
- ChatBadge와 그에 대응하는 ChatBadgeConfiguration을 구현합니다.

### 미비된 작업

- ExampleView가 아직 준비되지 않았습니다. #57 PR이 병합되면 해당 PR의 방식에 따라 비슷한 형태의 ExampleView를 추가할 예정입니다.
- 이 PR에서는 아직 fgYellowNormal이 fgOliveNormal을 바라보고 있기 때문에 #63 PR이 병합되어야 뷰의 색상이 제대로 보이게 됩니다

## 스크린샷

| | StatusBadge | ChatBadge|
|---|---|---|
| LightMode | <img width="474" alt="스크린샷 2024-08-09 오전 11 07 48" src="https://github.com/user-attachments/assets/a8335a71-8aa9-4f09-afbc-e8cde952a35b"> | <img width="188" alt="스크린샷 2024-08-09 오전 11 07 30" src="https://github.com/user-attachments/assets/dd764dc8-9b8a-462c-bd31-526cff62941b"> |
| DarkMode | <img width="468" alt="스크린샷 2024-08-09 오전 11 08 10" src="https://github.com/user-attachments/assets/1af79368-3412-41da-8184-219f9c164182"> | <img width="305" alt="스크린샷 2024-08-09 오전 11 08 03" src="https://github.com/user-attachments/assets/0fc7319a-cce9-480c-b0ab-58f697e42ccf"> |

bonoogi committed:
[Feature] StatusBadge, ChatBadge 구현

solchan87 commented on BezierSwift/Sources/Components/Badges/ChatBadge.swift:
```diff
@@ -0,0 +1,67 @@
+
+import SwiftUI
+
+private enum Constant {
```
Color 쓰는 목적이 다 달라서요. 컬러까지는 Constant를 쓰지 않고 있어요. 요건 바로 입력해줘도 좋을 것 같아요.

solchan87 commented on BezierSwift/Sources/Components/Badges/ChatBadge.swift:
```diff
@@ -0,0 +1,67 @@
+
+  public var body: some View {
+    ZStack(alignment: .center) {
+      BezierIcon
+        .lock
+        .image
```
ViewModifier와 구분지어주기 위해서 image까지는 한줄로 써주는 게 좋을 것 같아요
```suggestion
      BezierIcon.lock.image
```

solchan87 commented on BezierSwift/Sources/Components/Badges/ChatBadge.swift:
```diff
@@ -0,0 +1,67 @@
+      BezierIcon
+        .lock
+        .image
+        .frame(width: self.configuration.iconLength, height: self.configuration.iconLength)
```
모바일 버튼 PR에  length 함수가 있는데 버튼 PR 머지 이후에 한번 리베이스 하면 좋을 것 같아요

solchan87 commented on BezierSwift/Sources/Components/Badges/ChatBadge.swift:
```diff
@@ -0,0 +1,67 @@
+        .foregroundColor(Constant.iconColor.color)
+    }
+    .frame(width: self.configuration.length, height: self.configuration.length)
+    .background(
```
ZStack 선언후에 백그라운드가 들어간 것 같아서요. Circle이 ZStack 안으로 들어가는게 좋을 것 같아요

solchan87 commented on BezierSwift/Sources/Components/Badges/ChatBadge.swift:
```diff
@@ -0,0 +1,67 @@
+    .background(
+      Circle()
+        .foregroundColor(Constant.backgroundColor.color)
+    )
```
컴포넌트에 오퍼시티가 들어갈 수 있어서요. 컴포넌트가 끝나는 지점에 .compositingGroup() 선언 해주시면 감사합니다!
https://desk.channel.io/root/groups/iOS-52635/65cad70a40a0c7b277b5

bonoogi replied:
이거 정말 까먹기 쉽네요. BaseView같은걸로 통합하는게 나으려나... 🤔

solchan87 commented on BezierSwift/Sources/Components/Badges/StatusBadge.swift:
```diff
@@ -0,0 +1,110 @@
+
+import SwiftUI
+
+private enum Constant {
```
요것도 컬러 바로 써주는게 좋을 것 같아요!

solchan87 commented on BezierSwift/Sources/Components/Badges/StatusBadge.swift:
```diff
@@ -0,0 +1,110 @@
+  }
+
+  public var body: some View {
+    ZStack(alignment: .center) {
```
위에 내용과 동일한데 ZStack, Background 둘중 하나만 쓰면 좋을 것 같아요.

solchan87 commented on BezierSwift/Sources/Components/Badges/ChatBadge.swift:
```diff
@@ -0,0 +1,67 @@
+    }
+    .frame(width: self.configuration.length, height: self.configuration.length)
+    .background(
+      Circle()
```
Shape 는 fill 을 써서 컬러 주입해주는 것이 좋을 것 같아요!
https://developer.apple.com/documentation/swiftui/shape/fill(style:)

solchan87 commented on BezierSwift/Sources/Components/Badges/StatusBadge.swift:
```diff
@@ -0,0 +1,110 @@
+    .frame(width: self.configuration.length, height: self.configuration.length)
+    .background(
+      Circle()
+        .foregroundColor(Constant.backgroundColor.color)
```
여기 컬러로 fill 이면 좋을 것 같아요!

solchan87 commented on BezierSwift/Sources/Components/Badges/StatusBadgeConfiguration.swift:
```diff
@@ -0,0 +1,55 @@
+    case medium
+  }
+
+  let isOnline: Bool
```
최대한 Figma랑 동일한 이름으로 가면 좋을 것 같은데 어떠신가요??

bonoogi replied:
[iOSGithub 쓰레드에서도 얘기했다시피](https://desk.channel.io/root/threads/groups/iOSGitHub-136453/66b57dfa5087ee49fab4/66b596eee8c76d374c0e) 피그마의 DnD 같은 경우에는 그대로 변수명으로 쓰기가 좀 그래서, doNotDisturb로 적용해두었습니다.

solchan87 reviewed (commented):
핀 작업 감사합니다! SwiftUi 관련해서 몇가지 리뷰 남겼어요! 두개 컴포넌트 스타일이 비슷해서 리뷰 반영하실 때 같이 봐주시면 감사합니다!

bonoogi committed:
[Etc] 톰 리뷰 반영

bonoogi committed:
[Fix] StatusBadge icon 뷰의 형태에 따라 색상을 채우는 방법이 달라지도록 처리

solchan87 commented on BezierSwift/Sources/Components/Badges/StatusBadge.swift:
```diff
@@ -0,0 +1,102 @@
+
+extension StatusBadge {
+  @ViewBuilder
+  private var icon: some View {
+    if self.configuration.doNotDisturb {
+      BezierIcon.moonFilled.image
+        .foregroundColor(self.configuration.iconColor.color)
+    } else {
+      Circle()
+        .fill(self.configuration.iconColor.color)
+    }
+  }
```
```suggestion
  private var icon: some View {
    if self.configuration.doNotDisturb {
      BezierIcon.moonFilled.image
    } else {
      Circle()
        .fill()
    }
  }
```
로 하고 외부에서 공통으로 foregroundColor 줘도 좋을 것 같아요

solchan87 reviewed (approved):
작업 감사합니다! 마이너한 코멘트 하나만 더 추가했습니다.

bonoogi committed:
[Etc] 톰 리뷰 반영

- View와 Shape이 혼용되서 쓰이는 부분에서 Shape에 fill을 달아 some View로 변환해 foregroundColor가 일관적으로 적용될 수 있도록 합니다.
